test/Microsoft.ML.Functional.Tests/Datasets/Iris.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+
+using System;
+using Microsoft.Data.DataView;
+using Microsoft.ML.Data;
+
+namespace Microsoft.ML.Functional.Tests.Datasets
+{
+    /// <summary>
+    /// A class for the Iris test dataset.
+    /// </summary>
+    internal sealed class Iris
+    {
+        [LoadColumn(0)]
+        public float Label { get; set; }
+
+        [LoadColumn(1)]
+        public float SepalLength { get; set; }
+
+        [LoadColumn(2)]
+        public float SepalWidth { get; set; }
+
+        [LoadColumn(4)]
+        public float PetalLength { get; set; }
+
+        [LoadColumn(5)]
+        public float PetalWidth { get; set; }
+
+        /// <summary>
+        /// The list of columns commonly used as features.
+        /// </summary>
+        public static readonly string[] Features = new string[] { "SepalLength", "SepalWidth", "PetalLength", "PetalWidth" };
+
+        public static IDataView LoadAsRankingProblem(MLContext mlContext, string filePath, bool hasHeader, char separatorChar, int seed = 1)
+        {
+            // Load the Iris data.
+            var data = mlContext.Data.ReadFromTextFile<Iris>(filePath, hasHeader: hasHeader, separatorChar: separatorChar);
+
+            // Create a function that generates a random groupId.
+            var rng = new Random(seed);
+            Action<Iris, IrisWithGroup> generateGroupId = (input, output) =>
+            {
+                output.Label = input.Label;
+                // The standard set used in tests has 150 rows
+                output.GroupId = (ushort)rng.Next(0, 30);
```

Comment:
>ushort [](start = 34, length = 6)

any reason why it's ushort?

Reply:
Remnants of when I was trying to read directly as a `KeyType`. (see #2642)

---
In reply to: [259509894](https://github.com/dotnet/machinelearning/pull/2646#discussion_r259509894) [](ancestors = 259509894)